osu.Game/Database/BeatmapDatabase.cs

```diff
@@ -112,7 +111,17 @@ public void Import(params string[] paths)
                     }
                 }
 
-                Import(new[] { beatmapSet });
+                try
+                {
+                    Import(beatmapSet);
+                }
+                catch
+                {
+                }
+                finally
+                {
+                    File.Delete(p);

```

Comment:
As far as I understand from [this comment](https://github.com/ppy/osu/pull/245#issuecomment-266920967), there are 2 possible cases when importing new beatmap:
1). The import operation was performed without any issues. We don't need .osz file anymore and can remove it.
2). The import operation failed. In this situation, the beatmap can be considered bad and be nuked with no remorse.